### ISSUE TITLE
Check for local duplicates of patient name + date of birth

### ISSUE THREAD

**Issue by arjunSussol** [openmsupply/mobile#5132]
## Is your feature request related to a problem? Please describe.

Trello: https://trello.com/c/EB4F1nRk/108-414a-warn-user-that-duplicate-patient-records-exist

They want the system to check that there are no local duplicate patients on the tablet before allowing the user to create a new patient. 

This verification would be done by checking for duplicates of 'patient name + date of birth'.

## Describe the solution you'd like

 1) Just show a warning when duplicate patients are detected but still allow the user to proceed then it will be fine to add this functionality for all clients

2) Ignore the lookup patient interference, client is just wanting to try and reduce the number of duplicates that are being created, I don't think they're expecting to eliminate the chance of them being created entirely.

## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

## Additional context

**Add any other context or screenshots about the feature request here.**

Jobcode: KIY:MOB

**Comment by arjunSussol** [comment]
Already merged, so closed!

**Comment by adamdewey** [comment]
@arjunSussol , @AnushaUp - this looks pretty good in general, but there is a bug with the user workflow - after the user has been warned about a duplicate patient they cannot continue to find and select this existing patient record to continue using it. 

I will raise a separate issue..

**Comment by adamdewey** [comment]
@arjunSussol @AnushaUp - actually I found a few bugs with the workflows, so we need to look at this before we can release it - I'll send more details in Telegram, cheers.

**Comment by AnushaUp** [comment]
> [x] If you click `Save` then duplicate patient will be created having same details as above.

It will let you save duplicate patient but give alert in the middle of entering information and directly save the patient without letting user to complete it.
